libr/anal/p/anal_mips_cs.c

```diff
@@ -758,32 +758,40 @@ static int analop(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *buf, int len,
 	}
 	op->id = insn->id;
 	opsize = op->size = insn->size;
+	op->refptr = 0;
 	switch (insn->id) {
 	case MIPS_INS_INVALID:
 		op->type = R_ANAL_OP_TYPE_ILL;
 		break;
 	case MIPS_INS_LB:
 	case MIPS_INS_LBU:
 	case MIPS_INS_LBUX:
+		op->refptr = 1;
+		 /* fallthrough */ 
 	case MIPS_INS_LW:
 	case MIPS_INS_LWC1:
 	case MIPS_INS_LWC2:
 	case MIPS_INS_LWL:
 	case MIPS_INS_LWR:
 	case MIPS_INS_LWXC1:
+		if (!op->refptr) {
+			op->refptr = 4;
+		}
+		 /* fallthrough */ 
 	case MIPS_INS_LD:
 	case MIPS_INS_LDC1:
 	case MIPS_INS_LDC2:
 	case MIPS_INS_LDL:
 	case MIPS_INS_LDR:
 	case MIPS_INS_LDXC1:
 		op->type = R_ANAL_OP_TYPE_LOAD;
-		op->refptr = 4;
+		if (!op->refptr) {
+			op->refptr = 8;
```

Comment:
uhm.. shuoldnt this be 4?

Reply:
ptr = "load memory of refptr bytes"

In MIPS® Architecture For Programmers Volume II-A:
"The contents of the 64-bit doubleword at the memory location specified by the aligned effective address are fetched
and placed in GPR rt. The 16-bit signed offset is added to the contents of GPR base to form the effective address"

So I think the refptr value is related to the instruction.

Reply:
i have added the test：

NAME=mips ld instruction
FILE=bins/elf/analysis/mips64r2-busybox
CMDS=<<EOF
s 0x120004460
pd 1
EOF
EXPECT=<<EOF
            0x120004460      208084df       ld a0, -main(gp)           ;# [0x1200ee290:8]=0x120005074 sym.main
EOF
RUN

Before this bug is fixed, the actual output is：
            0x120004460      208084df       ld a0, -0x7fe0(gp)         ; #[0x1200ee290:4]=0x20005074

Obviously something went wrong

Reply:
Thanks for clarifying, you are right